react/features/toolbox/components/AudioRouteButton.js

```diff
@@ -0,0 +1,160 @@
+// @flow
+
+import React, { Component } from 'react';
+import {
+    findNodeHandle,
+    requireNativeComponent,
+    NativeModules,
+    View
+} from 'react-native';
+import { connect } from 'react-redux';
+
+import { openDialog } from '../../base/dialog';
+import { AudioRoutePickerDialog } from '../../mobile/audio-mode';
+
+import ToolbarButton from './ToolbarButton';
+
+/**
+ * Define the {@code MPVolumeView} React component. It will only be available
+ * on iOS.
+ */
+let MPVolumeView;
+
+if (NativeModules.MPVolumeViewManager) {
+    MPVolumeView = requireNativeComponent('MPVolumeView', null);
+}
+
+/**
+ * Style required to hide the {@code MPVolumeView} view, since it's displayed
+ * programmatically.
+ */
+const HIDE_VIEW_STYLE = { display: 'none' };
+
+type Props = {
+
+    /**
+     * Used to show the {@code AudioRoutePickerDialog}.
+     */
+    dispatch: Function,
+
+    /**
+     * The name of the Icon of this {@code AudioRouteButton}.
+     */
+    iconName: string,
+
+    /**
+     * The style of the Icon of this {@code AudioRouteButton}.
+     */
+    iconStyle: Object,
+
+    /**
+     * {@code AudioRouteButton} styles.
+     */
+    style: Array<*> | Object,
+
+    /**
+     * The color underlying the button.
+     */
+    underlayColor: string
+};
+
+/**
+ * A toolbar button which triggers an audio route picker when pressed.
+ */
+class AudioRouteButton extends Component<Props> {
+    _volumeComponent: ?Object;
+
+    /**
+     * Indicates if there is support for audio device selection via this button.
+     *
+     * @returns {boolean} - True if audio device selection is supported, false
+     * otherwise.
+     */
+    static supported() {
```

Comment:
If `AudioRouteButton` is not `supported()`, then why is it exported? It seems more complex than it needs to be.